src/api/trace/span.rs

```diff
@@ -38,6 +39,41 @@ pub trait Span: fmt::Debug + 'static + Send + Sync {
         self.add_event_with_timestamp(name, SystemTime::now(), attributes)
     }
 
+    /// Convenience method to record an exception/error as an `Event`
+    ///
+    /// An exception SHOULD be recorded as an Event on the span during which it occurred.
+    /// The name of the event MUST be "exception".
+    ///
+    /// The semantic conventions for Errors are described in ["Semantic Conventions for Exceptions"](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/exceptions.md)
+    ///
+    /// For now we will not set `exception.stacktrace` attribute since the `Error::backtrace`
+    /// method is still in nightly. User can provide stacktrace by using `record_exception_with_stacktrace`
+    /// method.
+    ///
+    /// User can custom exception message by overriding `fmt::Display` trait's `fmt` method for error.
+    fn record_exception(&self, err: &dyn Error) {
+        let attributes = vec![api::KeyValue::new(
+            "exception.message",
+            err.to_string(),
+        )];
+
+        self.add_event("exception".to_string(), attributes);
+    }
+
+    /// Convenience method to record a exception/error as an `Event` with custom stacktrace
+    ///
+    /// See `Span:record_exception` method for more details.
+    fn record_exception_with_stacktrace(&self, err: &dyn Error, stacktrace: String) {
+        let mut attributes = vec![api::KeyValue::new(
+            "exception.message",
+            err.to_string(),
+        )];
+
+        attributes.push(api::KeyValue::new("exception.stacktrace", stacktrace));
```

Comment:
This should be included in the `vec!` macro call. Otherwise the attributes vec will have an initial capacity of one, and the call here to `push` will have to extend the capacity up to 4. If you include both `vec![k1, k2]`, then it will be a single heap allocation and the vec's capacity will be 2.

Reply:
Sorry I should have considered that.